### PR TITLE
Production release v1.0.1

### PR DESCRIPTION
**Pull request by patrickvinograd**
* Avoid fetching full result set when doing filter counts

* Limit autocomplete query

* Update spec to account for autocomplete limit

### CODE REVIEW

**Comment by patrickvinograd** [comment]
This ought to cut down on low-memory alerts in production.

File: spec/models/institution_spec.rb [hunk 1]
```diff
@@ -134,10 +134,10 @@
       end
     end
 
-    it "nil or empty returns all institutions" do
+    it "nil or empty returns up to max returned institutions" do
       [nil, "", "      "].each do |arg|
         inst = Institution.autocomplete(arg)
-        expect(inst.size).to eql(Institution.all.count)
+        expect(inst.size).to eql([Institution::AUTOCOMPLETE_MAX, Institution.all.count].min)
```
Comment (jkassemi) [inline spec/models/institution_spec.rb]: Why not just use `Institution::AUTOCOMPLETE_MAX`?

Comment (patrickvinograd) [inline spec/models/institution_spec.rb]: The mock Institution database had 41 entries...close enough that if we decided to tweak the autocomplete_max upwards, it would be less than the max and require updating the spec.

**Comment by saintsoup52** [comment]
So can someone explain to me what is happening here in layman's terms?

I am anxious that the results of a search for a city like Los Angeles or New York will only yield 41 results.

If we are only talking about the autocomplete drop down then that is fine, but I would have serious concerns if we only showing 41 results for a "Los Angeles" search when it should return 100+.

**Comment by patrickvinograd** [comment]
The autocomplete drop down is now limited to 40 results. This still gives a pretty good list of choices for autocomplete without fetching thousands of results that start with "uni" for example. 

When an actual search is submitted, the behavior is *exactly the same as before*, except that we are being more efficient about how we are querying the database to avoid an unnecessary fetch of all results just to count them.

**Comment by saintsoup52** [comment]
Phew... that all sounds reasonable to me.
